src/mpl2/src/clusterEngine.cpp

```diff
@@ -560,6 +569,28 @@ void ClusteringEngine::createDataFlow()
   }
 }
 
+bool ClusteringEngine::stdCellsHaveLiberty()
+{
+  int valid_lib_cell_count = 0;
+  for (odb::dbInst* inst : block_->getInsts()) {
+    odb::dbMaster* master = inst->getMaster();
+    if (isIgnoredMaster(master) || master->isBlock()) {
+      continue;
+    }
+
+    const sta::LibertyCell* liberty_cell = network_->libertyCell(inst);
+    if (liberty_cell) {
+      ++valid_lib_cell_count;
+    }
+  }
+
+  if (valid_lib_cell_count == 0) {
+    return false;
+  }
+
+  return true;
```

Comment:
The logic boils down to this
```suggestion
      return true;
    }
  }
  return false;
```

Reply:
Thanks @rovinski !